The project is archived at github since 2022-03-25 so we cannot report this issue upstream.

--- pytest-tempdir-2019.10.12/pytest_tempdir/plugin.py.orig
+++ pytest-tempdir-2019.10.12/pytest_tempdir/plugin.py
@@ -147,7 +147,7 @@
     TempDir(config)
 
 
-@pytest.mark.trylast
+@pytest.hookimpl(trylast=True)
 def pytest_tempdir_temproot():
     return py.path.local.get_temproot()  # pylint: disable=no-member
 
